jplag/src/main/java/de/jplag/reporting/jsonfactory/DirectoryManager.java

```diff
@@ -0,0 +1,88 @@
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.Comparator;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DirectoryManager {
+    private static final Logger logger = LoggerFactory.getLogger(DirectoryManager.class);
+
+    /**
+     * Creates a directory.
+     * @param path The path under which the new directory ought to be created
+     * @param name The name of the new directory
+     * @return The created directory
+     */
+    public static File createDirectory(String path, String name) throws IOException {
+        File directory = new File(path.concat("/").concat(name));
+        if (!directory.exists() && !directory.mkdirs()) {
+            throw new IOException("Failed to create dir.");
+        }
+        return directory;
+    }
+
+    /**
+     * Create a directory with the given path
+     * @param path The path of the new directory
+     */
+    public static void createDirectory(String path) throws IOException {
+        createDirectory(path, "");
+    }
+
+    /**
+     * Delete the directory and all of its contents, identified by the given path
+     * @param path The path that identifies the directory to delete
+     */
+    public static void deleteDirectory(String path) {
+        try (var f = Files.walk(Path.of(path))) {
+            f.sorted(Comparator.reverseOrder()).map(Path::toFile).forEach(File::delete);
+        } catch (IOException e) {
+            logger.error("Could not delete folder " + path, e);
+        }
+    }
+
+    /**
+     * Zip the directory identified by the given path
+     * @param path The path that identifies the directory to zip
+     * @return True if zip was successful, false otherwise
+     */
+    public static boolean zipDirectory(String path) {
+        Path p = Path.of(path);
+        String zipName = path + ".zip";
+
+        try (ZipOutputStream zos = new ZipOutputStream(new FileOutputStream(zipName))) {
+
+            Files.walkFileTree(p, new SimpleFileVisitor<>() {
+                @Override
+                public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                    Path targetFile = p.relativize(file);
+                    zos.putNextEntry(new ZipEntry(targetFile.toString()));
+
+                    byte[] bytes = Files.readAllBytes(file);
+                    zos.write(bytes, 0, bytes.length);
+                    return FileVisitResult.CONTINUE;
+                }
+
+                @Override
+                public FileVisitResult visitFileFailed(Path file, IOException exc) {
+                    logger.error("Unable to zip " + file, exc);
+                    return FileVisitResult.CONTINUE;
+                }
+            });
+        } catch (IOException e) {
+            return false;
+        }
```

Comment:
The documentation was a bit incomplete for me, so my suggestions may already be covered by the current implementation.
1) If visiting a file failed, I would say the zipping failed entirely. Otherwise, we lose information. Thus, I would guess we need to return `FileVisitResult.TERMINATE` here. However, I did not find in the documentation whether that triggers an exception or how to otherwise get informed that zipping was terminated.
2) If the file walking fails after already having visited some files, are there already files written on disk or is some /tmp/ intermediate file used? Otherwise, we should (try to) cleanup when catching the `IOException`.

Reply:
>If visiting a file failed, I would say the zipping failed entirely. Otherwise, we lose information. Thus, I would guess we need to return FileVisitResult.TERMINATE here. However, I did not find in the documentation whether that triggers an exception or how to otherwise get informed that zipping was terminated.

As the report viewer currently does not handle missing files very well, I agree. But once the report viewer is able to recognise that files are missing and indicate it to the user, we could very well use `CONTINUE` instead of `TERMINATE`.
Using `TERMINATE` does not trigger an exception but simply stops the visiting process.

> If the file walking fails after already having visited some files, are there already files written on disk or is some /tmp/ intermediate file used? Otherwise, we should (try to) cleanup when catching the IOException.

I could not locate any hidden files. But an incomplete zip folder remains, so I chose to throw an exception when a visit fails, so that the catch code is triggered and the zip folder is deleted. The unzipped version will remain.